src/helpers/cards.ts

```diff
@@ -763,4 +766,83 @@ export const updateCardSettings = async (
   return settings;
 };
 
+export const createCardSpendingLimit = async (
+  req,
+  res
+): Promise<CardSpendingLimitControl> => {
+  const { scope_id: cardId, limit, idempotency_key: idempotencyKey } = req.body;
+  const cardData = await db.getCardData(cardId);
+
+  if (!cardData) {
+    return res.status(HttpStatusCodes.NOT_FOUND).send({
+      errors: [
+        {
+          id: uuid.v4(),
+          status: HttpStatusCodes.NOT_FOUND,
+          code: "model_not_found",
+          title: "Model Not Found",
+          detail: `Couldn't find 'Solaris::CardAccount' for id '${cardId}'.`,
+        },
+      ],
+    });
+  }
+
+  const cardControl = cardData.controls?.find(
```

Comment:
I am not sure if the `idempotency_key` check is enough to understand if the limit is already created.

Reply:
I've tried on sandbox and solaris continue pushing same limit into the card control. They just use the lowest during validation